packages/gestalt/src/SideNavigation/NavigationContent.js

```diff
@@ -0,0 +1,149 @@
+// @flow strict
+import { type Node as ReactNode, useEffect, useRef, useState } from 'react';
+import classnames from 'classnames';
+import Collapser from './Collapser';
+import {
+  countItemsWithIcon,
+  groupIconlessChildren,
+  validateChildren,
+} from './navigationChildrenUtils';
+import borderStyles from '../Borders.css';
+import Box from '../Box';
+import boxStyles from '../Box.css';
+import { useSideNavigation } from '../contexts/SideNavigationProvider';
+import Divider from '../Divider';
+import Flex from '../Flex';
+import layoutStyles from '../Layout.css';
+import { type Props as SideNavigationProps } from '../SideNavigation';
+import styles from '../SideNavigation.css';
+import { flattenChildrenWithKeys } from '../utils/flattenChildren';
+import { FixedZIndex } from '../zIndex';
+
+type Props = { ...SideNavigationProps };
+type Timeout = ReturnType<typeof setTimeout>;
+
+export default function NavigationContent({
+  accessibilityLabel,
+  children,
+  footer,
+  header,
+  showBorder,
+}: Props): ReactNode {
+  const navigationChildren = flattenChildrenWithKeys(children);
+
+  validateChildren({ children: navigationChildren, filterLevel: 'main' });
+
+  const {
+    collapsible,
+    collapsed: sideNavigationCollapsed,
+    onCollapse,
+    overlayPreview,
+    setOverlayPreview,
+    transitioning,
+  } = useSideNavigation();
+
+  const scrollContainer = useRef<HTMLDivElement | null>(null);
+  const [isScrolled, setIsScrolled] = useState(false);
+  const previewTimeoutRef = useRef<?Timeout>();
+
+  useEffect(() => {
+    const element = scrollContainer.current;
+    const scrollHandler = () => setIsScrolled(!!element?.scrollTop);
+
+    const mouseEnterHandler = () => {
+      if (sideNavigationCollapsed && !transitioning) {
+        clearTimeout(previewTimeoutRef.current);
+        setOverlayPreview(true);
+      }
+    };
+
+    const mouseLeaveHandler = () => {
+      if (sideNavigationCollapsed) {
```

Comment:
sideNavigationCollapsed if it's collapsed... how is that we are closing it?

Reply:
`sideNavigationCollapsed` means it is in collapsed state even when it is expanded for preview. Preview and collapsed are two different things. Preview state cannot happen unless the sidenav is collapsed.
Here the logic applied only when sidenav is collapsed but the timeout is related to preview state.